framework/decode/vulkan_replay_consumer_base.cpp

```diff
@@ -2084,8 +2089,16 @@ VkResult VulkanReplayConsumerBase::CreateSurface(InstanceInfo*
     if ((options_.surface_index == -1) || (options_.surface_index == create_surface_count_))
     {
         // Create a window for our surface.
-        Window* window = window_factory_->Create(
-            kDefaultWindowPositionX, kDefaultWindowPositionY, kDefaultWindowWidth, kDefaultWindowHeight);
+        assert(application_);
```

Comment:
Is it actually possible to get to this point in this function (and the others in vulkan_replay_consumer_base.cpp that check wsi_context and window_factory with assert and/or ternary operator) without wsi_context and window_factory being valid?

Reply:
It _shouldn't_ be possible to get there with nulls...since null derefs are guarded here, if they are null it'll cause errors later down the line (still shouldn't crash though, just spit out an error message and bail).  The `assert()` is just to make sure that if I made any mistakes in the logic leading up to there it will be instantly obvious.